src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/PniService.kt

```diff
@@ -68,8 +68,9 @@ class PniService(
     val assessmentId = assessmentIdDate.first
 
     // section 6
-    var completedAssessmentWithSara = oasysService.getAssessmentWithCompletedSara(prisonNumber) ?: assessmentId
-    val relationships = oasysService.getRelationships(completedAssessmentWithSara)
+    var completedAssessmentIdWithSara = oasysService.getAssessmentWithCompletedSara(prisonNumber)
+    val relationshipsForCompletedAssessmentWithSara = completedAssessmentIdWithSara?.let { oasysService.getRelationships(it) }
+    val relationships = oasysService.getRelationships(assessmentId)
```

Comment:
Do you want to move like 72 below 73

Its better to check if the relationships has sara information, before trying to find it from previous assessment ids

Reply:
Have refactored the code to included checking for an existing SARA before going back in time to search for one.